src/adaptors/hyperdrive/index.js

```diff
@@ -0,0 +1,324 @@
+    const virtualAssets = 1;
+    const virtualShares = 1e6;
+    const vaultSharePrice = (totalSupplyAssets + virtualAssets) / (totalSupplyShares + virtualShares) * 1e12;
+
+    vaultSharesBalance = position.supplyShares / 1e6 * vaultSharePrice;
+  } else if (config.vaultSharesToken !== '0x0000000000000000000000000000000000000000') {
+    vaultSharesBalance = (
+      await sdk.api.erc20.balanceOf({
+        target: config.vaultSharesToken,
+        owner: poolContract.address,
+        chain: config.chain
+      })
+    ).output;
+  } else {
+    // Use base token balance as vault shares balance
+    vaultSharesBalance = baseTokenBalance;
+  }
+
+  return vaultSharesBalance;
+}
+
+function encodeMorphoMarketIds(baseToken, collateral, oracle, irm, lltv) {
+  const packedIds = ethers.utils.defaultAbiCoder.encode(
+    ['address', 'address', 'address', 'address', 'uint256'],
+    [baseToken, collateral, oracle, irm, lltv]
+  );
+  return ethers.utils.keccak256(packedIds);
+}
+
+async function getApy(chain) {
+  const { registry = FUTURE_REGISTRY_ADDRESS } = config[chain];
+
+  try {
+    // First get the number of instances
+    const numInstances = (
+      await sdk.api.abi.call({
+        target: registry,
+        chain,
+        abi: 'function getNumberOfInstances() view returns (uint256)',
+      })
+    ).output;
+
+    // Then fetch each instance
+    const instanceCalls = Array.from({ length: Number(numInstances) }, (_, i) => ({
+      target: registry,
+      params: [i],
+    }));
+
+    let instances = (
+      await sdk.api.abi.multiCall({
+        abi: 'function getInstanceAtIndex(uint256) view returns (address)',
+        calls: instanceCalls,
+        chain,
+      })
+    ).output.map(o => o.output);
+
+    const poolNames = (
+      await sdk.api.abi.multiCall({
+        abi: 'function name() view returns (string)',
+        calls: instances.map(i => ({ target: i })),
+        chain
+      })
+    ).output.map(o => o.output);
+
+    const poolConfig = (
+      await sdk.api.abi.multiCall({
+        abi: GET_POOL_CONFIG_ABI,
+        calls: instances.map(i => ({ target: i })),
+        chain
+      })
+    ).output.map(o => o.output);
+
+    // Add chain to each config
+    poolConfig.forEach(config => {
+      config.chain = chain;
+    });
+
+    // Get token addresses and fetch prices
+    let priceWithBase = false;
+    let tokenAddress;
+    await Promise.all(poolConfig.map(async config => {
+      tokenAddress = config.vaultSharesToken === "0x0000000000000000000000000000000000000000" 
+        ? config.baseToken 
+        : config.vaultSharesToken;
+      let priceKey = `${chain}:${tokenAddress}`;
+      config.token_contract_address = tokenAddress;
+      let priceResponse = await axios.get(`https://coins.llama.fi/prices/current/${priceKey}`);
+      let price = priceResponse.data.coins[priceKey];
+      if (price === undefined && config.baseToken !== '0x0000000000000000000000000000000000000000') {
```

Comment:
nit: add some comment explaining this edge case

Reply:
done in c3a165eb10f3c18fc73249bab2f0f9ffeb99e688